src/main/java/org/apache/maven/plugins/war/util/PathSet.java

```diff
@@ -113,7 +109,7 @@ public void addAll( Collection<String> paths, String prefix )
     {
         for ( String val : paths )
         {
-            add( prefix + val );
+            add( Paths.get( prefix, val ).toString() );
```

Comment:
Same here

Reply:
Another detail:

PathSet class still store the set of paths as String due to serialization / deserialization to XML. This could not be changed to Set<Path>, otherwise you will get a StackOverflowError during this process.